apollo-federation/src/sources/connect/json_selection/parser.rs

```diff
@@ -1181,17 +1158,18 @@ pub(crate) fn parse_string_literal(input: Span) -> ParseResult<WithRange<String>
                 chars.push(c);
             }
 
-            if let Some(remainder) = remainder_opt {
-                Ok((
-                    remainder,
-                    WithRange::new(
-                        chars.iter().collect::<String>(),
-                        Some(start..remainder.location_offset()),
-                    ),
-                ))
-            } else {
-                Err(nom_fail_message(input, "Unterminated string literal"))
-            }
+            remainder_opt.map_or_else(
+                || Err(nom_fail_message(input, "Unterminated string literal")),
+                |remainder| {
+                    Ok((
+                        remainder,
+                        WithRange::new(
+                            chars.iter().collect::<String>(),
+                            Some(start..remainder.location_offset()),
+                        ),
+                    ))
+                },
+            )
```

Comment:
I also like the `ok` methods for options, though not sure if it clears anything up...

```suggestion
            remainder_opt.ok_or_else(|| nom_fail_message(input, "Unterminated string literal"))
            .map(|remainder| {
                (
                    remainder,
                    WithRange::new(
                        chars.iter().collect::<String>(),
                        Some(start..remainder.location_offset()),
                    )
                 )
            })
```

Reply:
I like it as well!